session.go

```diff
@@ -1480,14 +1475,19 @@ func (s *session) handleCloseError(closeErr *closeError) {
 		}()
 	}
 
+	statelessResetErr := &StatelessResetError{}
+	versionNegotiationErr := &VersionNegotiationError{}
+	recreateErr := &errCloseForRecreating{}
+	applicationErr := &qerr.ApplicationError{}
+	transportErr := &qerr.TransportError{}
```

Comment:
nit: for all these where you declare the error pointers separately, you don't need to actually initialize an error:

```
var someErr *SomeType
if errors.As(err, &someErr) { ... }
```

You do need the `&SomeType{}` for `if stmt; cond { ... }`, because that form doesn't accept a `var name Type`.

Reply:
Fixed.